src/2d/shallow/surge/explicit_storm_module.f90

```diff
@@ -0,0 +1,506 @@
+        real(kind=8) :: t_next, t_prev
+        real(kind=8), allocatable :: u_next(:,:)
+        real(kind=8), allocatable :: v_next(:,:)
+        real(kind=8), allocatable :: p_next(:,:)
+        real(kind=8), allocatable :: u_prev(:,:)
+        real(kind=8), allocatable :: v_prev(:,:)
+        real(kind=8), allocatable :: p_prev(:,:)
+        ! The values will be updated incrementally:
+        !  when t>t_next, replace older states with newer states
+        !  and read in next snapshot.
+        
+        ! These will be used during the interpolation step
+        !  for times between t_prev and t_next
+        real(kind=8) :: t
+        real(kind=8), allocatable :: u(:,:)
+        real(kind=8), allocatable :: v(:,:)
+        real(kind=8), allocatable :: p(:,:)
+
+        ! Keep track of how many snapshots have been read
+        integer :: last_storm_index
+
+        ! for compatibility with parent storm module (?)
+        ! Storm physics
+        real(kind=8) :: ambient_pressure = 101.3d3 ! 101300 Pascals
+        real(kind=8) :: rho_air = 1.3d0
+
+        ! Store the storm data file for repeated reading
+        character(len=90) :: data_path_root
+
+    end type explicit_storm_type
+
+    ! Internal tracking variables for storm
+    logical, private :: DEBUG = .true. 
+    !logical, private :: DEBUG = .false. 
+
+contains
+
+    ! Setup routine for the explicit model
+    ! Open data files, get parameters, allocate memory,
+    !  and read in first two time snapshots of storm data
+    subroutine set_explicit_storm(storm_data_path_root, storm, log_unit)
+
+        use geoclaw_module, only: coordinate_system
+        use amr_module, only: t0
+
+        implicit none
+
+        ! Subroutine I/O
+        character(len=*), optional :: storm_data_path_root
+        type(explicit_storm_type), intent(in out) :: storm
+        integer, intent(in) :: log_unit
+
+        ! Local storage
+        integer, parameter :: l_file = 701
+        integer :: i, j, k, io_status, num_rows, num_cols
+        real(kind=8) :: forecast_time,last_time,x(2),y(2),ds,dt,dx,dy
+        character(len=100) :: storm_data_path
+
+        ! Reading buffer variables
+        character(len=20000) :: dummy_line
+
+        ! Storm type only works on lat-long coordinate systems
+        if (coordinate_system /= 2) then
+            stop "explicit storm type does only works on lat-long coordinates."
+        endif
+
+        ! We need to count two things:
+        !   number of latitude coords (ny)
+        !   number of longitude coords (nx)
+        ! We don't need number of time snapshots (nt)
+        !   since we will load snapshots only as needed
+        ! The datafiles for u, v, p contain nx*ny*nt values
+
+        ! Open latitudes data file
+        if (present(storm_data_path_root)) then
+            storm%data_path_root = storm_data_path_root
+        else
+            storm%data_path_root = './t1330-case0550-'
```

Comment:
Probably should just give this as the current working directory if not specified.